### PR TITLE
Simple multi line support

### PR DESCRIPTION
**Pull request by Uight**
Just putting it out there. I think this is one of the best solutions to enable basic multilien support for different definitions and comments. Also supports to some extend multiple definitions per line.
I think the only "really" better way would be to switch the parser in its entirety like the milestones/v1.8 branch. 

Still think this solution could solve some problems. 
Basically this parser uses the nextlineprovider to split lines based on the prefixes for definitions.
it searches for line termiantions ";" or new lines to split definitions.

Basically for what it is its doing a good job. I also fixed some stuff that was criticized when we talked about this in one issue.
The line number when an error occures is now pretty accurate. About the same as the vector candb++ errors which might also be one line off at some points. 
Missing ";" are not longer fixed as that is not done by candb++ which failes parsing if a required ; is not present.

### CODE REVIEW

**Comment by Uight** [comment]
@Adhara3 @EFeru or @Whitehouse112 is it worth merging this with main or just delte the branch and PR?
Also its seems repo is inacvtive for a while now. I would have some time on hand now again to do stuff like extended multiplexing and also add that to message parsing. Worth it?

**Comment by Adhara3** [comment]
Hi, I personally do not like the solutions that _partially_ solve an issue, but the discussions we had in the past did not bring to a shared agreement nor my proposal was accepted.

So I leave @EFeru who is the owner to choose.

Best
A

**Comment by Uight** [comment]
@Adhara3 whitehouse allready fixed ine think in main with ; in the text. I gues thats enough for now

**Comment by Adhara3** [comment]
This is a neverending discussion. I still think the parser itself has reached a limit and should be rewritten. 
Nevertheless, no problem for me going on with a patched solution instead.

Worst case scenario, I'll fork.

Cheers
A

**Comment by EFeru** [comment]
Hi @Uight, @Adhara3 ,

Can you point me to the dilemma we have and the limitation we have that requires the re-writing of the parser?

@Uight, I see you put a lot of effort into this PR. What smaller chunks can be re-used in separate PR(s) that are easy to merge? I find it hard that whole PR is thrown away.
